### DISCUSSION TITLE
OpenFAST simulation without blades

### DISCUSSION THREAD

**Discussion by relhamoud** [OpenFAST/openfast#2040] [Q&A]
Hello everyone,

I am interested in simulating OpenFAST without blades. When i set NumBl = 0, it gives an error and it displays that NumBl should be positive.

Is there any trick to do it in order to run a simulation without any blades ?

Thank you for your help.

Best Regards,

Riad

**Comment by jjonkman** [comment]
While OpenFAST doesn't have an option to specify `NumBl` = 0, you could avoid modeling blades by setting the input parameters such that the effect of the blades is negligible, such as by disabling the blade degrees of freedom in ElastoDyn and setting the blade lengths to zero by setting `TipRad` = `HubRad`.

The "Ideal_Beam" tests in the OpenFAST r-test (https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast) may be good examples as the parameters are set so that OpenFAST models only a single uniform beam. 

Best regards,

**Reply by juleshd** [reply]
Hello @jjonkman,

I am currently trying to compute the loads in the moorings for a regular floating platform, anchored to the seabed. I took as a starting example the case "5MW_OC6Semi_MD_Linear" and I am trying to adapt it for my need. Next step is to remove the blades, the hub and the tower.
I found this previous discussion about a simulation "without blades" and I am looking to set the TipRad=HubRad like you said. If I set it up like this, a first error tells me that TipRad must be greater than HubRad. I then set TipRad to 1e-5 and HubRad to 0. Another error shows: 

FAST_InitializeAll:ED_Init:ED_ReadInput:ReadBladeInputs:ReadBladeFile:The ElastoDyn Blade file,
.\../5MW_Baseline/NRELOffshrBsline5MW_Blade2.dat, DISTRIBUTED BLADE PROPERTIES table contains the
PitchAxis column.  This column is unused and will be removed in future releases
ED_Init:ED_AllocOutput:MeshCommit: Line2 element 1 has 0 length.
   n2 = n(2) = (0,1.83697E-21,6.00000E-05)
   n1 = n(1) = (0,9.18485E-22,5.50000E-05)

Yet, my ElastoDyn.dat file is almost like the one set for the "Ideal_Beam" case. If you have any idea than could help me for my case, i would really appreciate it.

Have a nice day,

Jules

**Reply by jjonkman** [reply]
Dear @juleshd,

OpenFAST has internal checks to prevent unphysical model setup.  To avoid modeling the blades by defining a blade of length epsilon (= `TipRad` - `HubRad`), you may have to play around with what epsilon is to get around the internal checks.  My guess is the "ideal_beam" r-test doesn't suffer the same error because in these r-test cases, the blade (beam) is modeled in BeamDyn instead of ElastoDyn via `CompElast` = 2.

Best regards,

**Reply by juleshd** [reply]
Dear @jjonkman,

Thanks for your help. Indeed if I set HubRad=0 and TipRad=7e-4, OpenFast goes through its internal checks and runs normally.

Best regards

**Comment by relhamoud** [comment]
Thank you @jjonkman for your reply and sorry for my late response.

I think i did not clarify very well the goal of simulating OpenFAST without blades. In fact, i am interested in studying the motion of the tower during blade installation process. However, the test you have suggested does not enable the tower degrees of freedom. Only the platform DoFs are enabled.

Best regards,

Riad

**Reply by jjonkman** [reply]
Dear @relhamoud,

I referenced the "Ideal_Beam" tests of  the OpenFAST r-test not for the platform or tower properties, but to show you how to set up ElastoDyn file so as to set the blade properties to zero-out their effect.  Of course, you can build your own example that has realistic tower and nacelle properties, just without blades.

Best regards,

**Reply by shijieQiu** [reply]
Dear @jjonkman ,
I try to run the case "Ideal_Beam_Free_Free_Linear", but I don't know why the following error message occurs:
OpenFAST-Unversioned from 4b6337fcffe859c5eeb5445deeef2046439e5152
Compile Info:
 - Compiler: Intel(R) Fortran Compiler 2021
 - Architecture: 64 bit
 - Precision: single
 - OpenMP: No
 - Date: Apr 05 2024
 - Time: 18:18:17
Execution Info:
 - Date: 04/09/2024
 - Time: 12:24:21+0800

OpenFAST input file heading:
    Free-free beam test case

Running ElastoDyn.
Nodal outputs section of ElastoDyn input file not found or improperly formatted.

FAST_InitializeAll:ED_Init:ED_AllocOutput:MeshCommit: Line2 element 1 has 0 length.
   n2 = n(2) = (0,0,5.00000E-05)
   n1 = n(1) = (0,0,2.50000E-05)


 OpenFAST encountered an error during module initialization.
 Simulation error level: FATAL ERROR

 Aborting OpenFAST.

Hope to get your reply, thank you

Best regards,

**Reply by jjonkman** [reply]
Dear @shijieQiu,

Did you change the OpenFAST input files for the case Ideal_Beam_Free_Free_Linear in any way?

Best regards
